satellite/metainfo/loop.go

```diff
@@ -163,18 +176,36 @@ waitformore:
 
 			// iterate over every segment in metainfo
 			for it.Next(ctx, &item) {
-				path := item.Key.String()
+				rawPath := item.Key.String()
 				pointer := &pb.Pointer{}
 
 				err = proto.Unmarshal(item.Value, pointer)
 				if err != nil {
 					return LoopError.New("unexpected error unmarshalling pointer %s", err)
 				}
 
-				nextObservers := observers[:0]
+				pathElements := storj.SplitPath(rawPath)
```

Comment:
@mniewrzal did we have a func to do this parsing instead of calling `storj.SplitPath`?